_unittest/conftest.py

```diff
@@ -197,13 +197,21 @@ def _method(
             test_project = None
         if not application:
             application = Hfss
-        return application(
-            project=test_project,
-            design=design_name,
-            solution_type=solution_type,
-            version=desktop_version,
-            non_graphical=NONGRAPHICAL,
-        )
+        if solution_type:
+            return application(
+                project=test_project,
+                design=design_name,
+                solution_type=solution_type,
+                version=desktop_version,
+                non_graphical=NONGRAPHICAL,
+            )
+        else:
+            return application(
+                project=test_project,
+                design=design_name,
+                version=desktop_version,
+                non_graphical=NONGRAPHICAL,
+            )
```

Comment:
```suggestion
        args = {
            project=test_project,
            design=design_name,
            version=desktop_version,
            non_graphical=NONGRAPHICAL,
        }
        if solution_type:
            args["solution_type"]=solution_type
        return application(**args)
```
Could we refactor into something like that ?